sdk/notificationhubs/notification-hubs/src/notificationHubsClient.ts

```diff
@@ -0,0 +1,419 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import {
+  BrowserPushChannel,
+  InstallationPatch,
+  InstallationType,
+  NotificationHubMessageType,
+  NotificationHubResponse,
+  NotificationHubsClientOptions,
+  PushHandleType,
+  SendOperationOptions,
+} from "./models";
+import {
+  HttpHeaders,
+  HttpMethods,
+  PipelineRequest,
+  PipelineResponse,
+  createHttpHeaders,
+  createPipelineRequest,
+} from "@azure/core-rest-pipeline";
+import { OperationOptions, ServiceClient } from "@azure/core-client";
+import {
+  createTokenProviderFromConnection,
+  parseNotificationHubsConnectionString,
+} from "./utils/connectionStringUtils";
+import { SasTokenProvider } from "@azure/core-amqp";
+import { tracingClient } from "./utils/tracing";
+
+const API_VERSION = "2020-06";
+
+/**
+ * This represents a client for Azure Notification Hubs to manage installations and send
+ * messages to devices.
+ */
+export class NotificationHubsClient extends ServiceClient {
+  private _baseUrl: string;
+  private _hubName: string;
+  private _sasTokenProvider: SasTokenProvider;
+
+  /**
+   * Creates a new instance of the NotificationClient with a connection string, hub name and options.
+   * @param connectionString - The Notification Hub Access Policy connection string.
+   * @param hubName - The name of the Azure Notification Hub.
+   * @param options - Options for configuring the Azure Notification Hubs client.
+   */
+  constructor(
```

Comment:
Currently, it is the only supported and probably will be for the next year or so.

Reply:
I wonder if there's something clever we can do here.... especially since the most common pattern beyond connection string is endpoint first, which unfortunately just looks like a string...